### PR TITLE
Change threads intialization flow

### PR DESCRIPTION
**Pull request by diega**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 342 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 211 major
* ![MINOR][MINOR] 102 minor
* ![INFO][INFO] 25 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [TxHandlerImpl.java#L32](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/net/handler/TxHandlerImpl.java#L32): Unused import - java.util.concurrent.ThreadFactory. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L49](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L49): Unused import - javax.annotation.PostConstruct. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L322](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L322): This call to "add()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L330](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L330): This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/Start.java#L117): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/Start.java#L117): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L110](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L110): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L133](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L133): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L42](https://github.com/rsksmart/rskj/blob/b80d2ac388dbc569ba625e836716bf92f8722f49/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L42): Constructor has 10 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/Start.java [hunk 1]
```diff
@@ -78,12 +95,26 @@ public Start(Rsk rsk,
...
         this.repository = repository;
+        this.blockchain = blockchain;
+        this.channelManager = channelManager;
     }
 
     public void startNode(String[] args) throws Exception {
         logger.info("Starting RSK");
 
         CLIInterface.call(rskSystemProperties, args);
+        logger.info("Running {},  core version: {}-{}", rskSystemProperties.genesisInfo(), rskSystemProperties.projectVersion(), rskSystemProperties.projectVersionModifier());
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/Start.java]: All of this is now logged under `start` instead of `general`, which makes so much sense!

**Comment by rskops** [comment]
SonarQube analysis reported 342 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 211 major
* ![MINOR][MINOR] 102 minor
* ![INFO][INFO] 25 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [TxHandlerImpl.java#L32](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/net/handler/TxHandlerImpl.java#L32): Unused import - java.util.concurrent.ThreadFactory. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L49](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L49): Unused import - javax.annotation.PostConstruct. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L322](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L322): This call to "add()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L330](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L330): This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/Start.java#L117): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/Start.java#L117): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L110](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L110): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L133](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L133): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L42](https://github.com/rsksmart/rskj/blob/ed760f91f1280c56fb763eab03c4596e41237f2d/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L42): Constructor has 10 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java [hunk 2]
```diff
@@ -89,13 +91,14 @@ public EthereumImpl(WorldManager worldManager,
...
-                Thread thread = new Thread(runnable);
+            peerServiceExecutor = Executors.newSingleThreadExecutor(runnable -> {
+                Thread thread = new Thread(runnable, "Peer Server");
                 thread.setUncaughtExceptionHandler((exceptionThread, exception) -> {
                     gLogger.error("Unable to start peer server", exception);
                 });
                 return thread;
-            }).execute(() -> peerServer.start(config.listenPort()));
+            });
+            peerServiceExecutor.execute(() -> peerServer.start(config.listenPort()));
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java]: I wonder why we're starting this here and not in `Start`.

File: rskj-core/src/main/java/org/ethereum/sync/SyncPool.java [hunk 3]
```diff
@@ -75,39 +76,39 @@
...
+            () -> {
+                try {
+                    heartBeat();
+                    processConnections();
+                    updateLowerUsefulDifficulty();
+                    fillUp();
+                    prepareActive();
+                } catch (Throwable t) {
+                    logger.error("Unhandled exception", t);
+                    panicProcessor.panic("syncpool", String.format("Unhandled exception {}", t.getMessage()));
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/sync/SyncPool.java]: Can you fix this log? I think it should be `panicProcessor.panic("syncpool", String.format("Unhandled exception %s", t.getMessage()));`

**Comment by rskops** [comment]
SonarQube analysis reported 338 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 211 major
* ![MINOR][MINOR] 100 minor
* ![INFO][INFO] 25 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L321](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L321): This call to "add()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L329](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L329): This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/Start.java#L117): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/Start.java#L117): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L110](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L110): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L133](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L133): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L42](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L42): Constructor has 10 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L117](https://github.com/rsksmart/rskj/blob/d8acf6f1640d0aa5e8b51a8f8362cc03d1e3e88d/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L117): Constructor new co.rsk.core.bc.BlockChainImpl(Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator, RskSystemProperties) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/Start.java [hunk 4]
```diff
@@ -163,6 +167,7 @@ public void startNode(String[] args) throws Exception {
             enablePeerDiscovery();
         }
 
+        messageHandler.start();
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/Start.java]: TL;DR;: I think the order is wrong and this should be started before `rsk.init()`.

In the old code, `@Bean public Rsk getRsk` depended on `NodeMessageHandler`. This means that `@Bean public NodeMessageHandler getNodeMessageHandler` was executed before `getRsk`. Which means that `messageHandler.start();` was executed before `rsk.init()`.
I don't necessarily think this is wrong, but let's try to maintain the same behavior.

File: rskj-core/src/main/java/co/rsk/Start.java [hunk 5]
```diff
@@ -168,6 +172,7 @@ public void startNode(String[] args) throws Exception {
         }
 
         messageHandler.start();
+        txHandler.start();
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/Start.java]: Same as before: `getNodeMessageHandler` depended on `TxHandler`, so `txHandler.start()` should go before `messageHandler.start()`

File: rskj-core/src/main/java/co/rsk/Start.java [hunk 6]
```diff
@@ -173,6 +173,7 @@ public void startNode(String[] args) throws Exception {
 
         messageHandler.start();
         txHandler.start();
+        channelManager.start();
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/Start.java]: `channelManager.start()` should go before `messageHandler.start()`. Whether it was executed before or after `txHandler.start()` was undefined, so you can do whatever sounds better.

File: rskj-core/src/main/java/co/rsk/util/RskCustomCache.java [hunk 7]
```diff
@@ -43,7 +43,14 @@ public Thread newThread(Runnable r) {
                 return new Thread(r, "BlockHeaderCacheTimer");
             }
         });
-        this.startCleanUp();
+    }
+
+    public void start() {
+        startCleanUp();
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/RskCustomCache.java]: Please inline the `startCleanUp` method to make things clearer.

File: rskj-core/src/main/java/co/rsk/metrics/HashRateCalculator.java [hunk 8]
```diff
@@ -39,6 +39,14 @@ public HashRateCalculator(BlockStore blockStore, RskCustomCache<ByteArrayWrapper
         this.headerCache = headerCache;
     }
 
+    public void start() {
+        headerCache.start();
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/metrics/HashRateCalculator.java]: It's weird to have this here but it's OK for now!

**Comment by rskops** [comment]
SonarQube analysis reported 335 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 208 major
* ![MINOR][MINOR] 100 minor
* ![INFO][INFO] 25 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L321](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L321): This call to "add()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![CRITICAL][CRITICAL] [ChannelManagerImpl.java#L329](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/org/ethereum/net/server/ChannelManagerImpl.java#L329): This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2250)
1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/Start.java#L117): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/Start.java#L117): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L110](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L110): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L133](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L133): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L42](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L42): Constructor has 10 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L117](https://github.com/rsksmart/rskj/blob/4e40ee9d0a675fdaf186f741457eec6e047e566f/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L117): Constructor new co.rsk.core.bc.BlockChainImpl(Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator, RskSystemProperties) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/Start.java (hunk 4 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/Start.java]: great catch, thanks!

**Pull request merged**